thanos/charts/alerts/alerts.yaml

```diff
@@ -0,0 +1,668 @@
+groups:
+- name: thanos-compact
+  rules:
+  - alert: ThanosCompactMultipleRunning
+    annotations:
+      description: No more than one Thanos Compact instance should be running at once.
+        There are {{$value}} instances running.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanoscompactmultiplerunning
+      summary: Thanos Compact has multiple instances running.
+    expr: sum by (job) (up{job=~".*thanos-compact.*"}) > 1
+    for: 5m
+    labels:
+      severity: warning
+  - alert: ThanosCompactHalted
+    annotations:
+      description: Thanos Compact {{$labels.job}} has failed to run and now is halted.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanoscompacthalted
+      summary: Thanos Compact has failed to run and is now halted.
+    expr: thanos_compact_halted{job=~".*thanos-compact.*"} == 1
+    for: 5m
+    labels:
+      severity: warning
+  - alert: ThanosCompactHighCompactionFailures
+    annotations:
+      description: Thanos Compact {{$labels.job}} is failing to execute {{$value |
+        humanize}}% of compactions.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanoscompacthighcompactionfailures
+      summary: Thanos Compact is failing to execute compactions.
+    expr: |
+      (
+        sum by (job) (rate(thanos_compact_group_compactions_failures_total{job=~".*thanos-compact.*"}[5m]))
+      /
+        sum by (job) (rate(thanos_compact_group_compactions_total{job=~".*thanos-compact.*"}[5m]))
+      * 100 > 5
+      )
+    for: 15m
+    labels:
+      severity: warning
+  - alert: ThanosCompactBucketHighOperationFailures
+    annotations:
+      description: Thanos Compact {{$labels.job}} Bucket is failing to execute {{$value
+        | humanize}}% of operations.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanoscompactbuckethighoperationfailures
+      summary: Thanos Compact Bucket is having a high number of operation failures.
+    expr: |
+      (
+        sum by (job) (rate(thanos_objstore_bucket_operation_failures_total{job=~".*thanos-compact.*"}[5m]))
+      /
+        sum by (job) (rate(thanos_objstore_bucket_operations_total{job=~".*thanos-compact.*"}[5m]))
+      * 100 > 5
+      )
```

Comment:
Did you verify if all the conditions work? Or will you iterate over it in QA?

Reply:
thanks, will go through them before QA. there is some changes needed indeed